src/Domain/EmbeddingMaps/ProductMaps.cpp

```diff
@@ -0,0 +1,133 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/EmbeddingMaps/ProductMaps.hpp"
+#include "ErrorHandling/Error.hpp"
+
+namespace EmbeddingMaps {
+
+template <size_t Dim1, size_t Dim2>
+ProductOf2Maps<Dim1, Dim2>::ProductOf2Maps(const EmbeddingMap<Dim1, Dim1>& map1,
+                                           const EmbeddingMap<Dim2, Dim2>& map2)
+    : EmbeddingMap<Dim1 + Dim2, Dim1 + Dim2>(),
+      map1_(map1.get_clone()),
+      map2_(map2.get_clone()) {}
+
+template <size_t Dim1, size_t Dim2>
+ProductOf2Maps<Dim1, Dim2>::ProductOf2Maps(
+    const ProductOf2Maps<Dim1, Dim2>& other)
+    : EmbeddingMap<Dim1 + Dim2, Dim1 + Dim2>(),
+      map1_(other.map1_->get_clone()),
+      map2_(other.map2_->get_clone()) {}
+
+template <>
+Point<2, Frame::Grid> ProductOf2Maps<1, 1>::operator()(
```

Comment:
I would say we put the definitions of the call operator and `inverse` in the header file and mark them as inline. I think this is possible and won't violate ODR